### PR TITLE
Error handling (EXPOSURERAPP-7508)

### PR DESCRIPTION
**Pull request by chiljamgossow**
ToDo 
- [X] server errors
- [X] test certificate decryption/decoding errors
- [ ] unit tests -> separate PR

### CODE REVIEW

File: Corona-Warn-App/build.gradle [hunk 1]
```diff
@@ -441,6 +441,4 @@ dependencies {
 
     // HCert
     implementation("com.upokecenter:cbor:4.4.1")
-
-    implementation("bouncycastle:bcprov-jdk16:136")
```
Comment (d4rken) [inline Corona-Warn-App/build.gradle:445]: Why was this previously added and now removed?

Comment (chiljamgossow) [reply Corona-Warn-App/build.gradle:445]: Works just fine without

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateApiV1.kt [hunk 2]
```diff
@@ -15,16 +15,24 @@ interface CovidCertificateApiV1 {
     @POST("/version/v1/publicKey")
     suspend fun sendPublicKey(
         @Body requestBody: PublicKeyUploadRequest
-    )
+    ): Response<Any>
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateApiV1.kt:18]: does it return anything ? would `Response<Unit>` work ?

**Review by mtwalli — DISMISSED** [review]
Lgtm

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateApiV1.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateApiV1.kt:18]: That's needs to be tested, what the server returns, Unit is likely fine

**Review by mtwalli — APPROVED** [review]

**Review by d4rken — APPROVED** [review]
lgtm

**Pull request merged**